### PR TITLE
WIP: Fixing the unittests for controller

### PR DESCRIPTION
**Pull request by nandajavarma**

### CODE REVIEW

**Comment by nandajavarma** [comment]
A lot of time was spent on doing this small thing. Well, learning experience I guess 🤕

**Comment by nandajavarma** [comment]
fip-commons PR [#9](https://github.com/sighupio/fip-commons/pull/9) and fip-results-controller PR [#7](https://github.com/sighupio/fip-results-controller/pull/7) are related to this.

File: internal/server/server_test.go [hunk 1]
```diff
@@ -125,90 +132,155 @@ func TestParseResponse(t *testing.T) {
...
+	if c.groupList == nil {
+		return nil, errors.New("doesn't exist")
+	}
+	return c.groupList, c.groupListErr
+}
+
+type resourceMapEntry struct {
+	list *metav1.APIResourceList
+	err  error
+}
```
Comment (angelbarrera92) [inline internal/server/server_test.go:199]: Does it makes sense to export this functions/structs to the fip-commons?

Comment (angelbarrera92) [reply internal/server/server_test.go:199]: Ping @nandajavarma

Comment (nandajavarma) [reply internal/server/server_test.go:199]: @angelbarrera92 sorry I haven't been getting back to this. I am not happy about these changes anymore and it has to change to handle unstructured data. Currently I am not sure if it is worth making this effort, considering the payoff is not much. I am moving this back to draft for now.

**Pull request closed without merging**